rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -254,50 +258,25 @@ private void call() {
         precompiledContract = precompiledContracts.getContractForAddress(new DataWord(targetAddress.getBytes()));
 
         if (precompiledContract != null) {
-            precompiledContract.init(tx, executionBlock, track, blockStore, receiptStore, result.getLogInfoList());
-            long requiredGas = precompiledContract.getGasForData(tx.getData());
-            BigInteger txGasLimit = toBI(tx.getGasLimit());
-
-            if (!localCall && txGasLimit.compareTo(BigInteger.valueOf(requiredGas)) < 0) {
-                // no refund
-                // no endowment
-                execError(String.format("Out of Gas calling precompiled contract 0x%s, required: %d, left: %s ",
-                        targetAddress.toString(), (requiredGas + basicTxCost), mEndGas));
-                mEndGas = BigInteger.ZERO;
-                return;
-            } else {
-                long gasUsed = requiredGas + basicTxCost;
-                mEndGas = txGasLimit.subtract(BigInteger.valueOf(requiredGas + basicTxCost));
-
-                // FIXME: save return for vm trace
-                try {
-                    byte[] out = precompiledContract.execute(tx.getData());
-                    result.setHReturn(out);
-                } catch (RuntimeException e) {
-                    result.setException(e);
-                }
+            precompiledContract.init(tx, executionBlock, cacheTrack, blockStore, receiptStore, result.getLogInfoList());
```

Comment:
So you changed:

```java
precompiledContract.init(tx, executionBlock, track, blockStore, receiptStore, result.getLogInfoList());
```

to:

```java
precompiledContract.init(tx, executionBlock, cacheTrack, blockStore, receiptStore, result.getLogInfoList());
```

I don't think this works for `Bridge` because it needs to update data.